K8SPSMDB-355 update doc how to scale cluster (#560)
percona/percona-server-mongodb-operator: source/scaling.rst

or down ensures new Pods are created and set to available Kubernetes
nodes.

The size of the cluster is controlled by the ``size`` key in the
:ref:`operator.custom-resource-options` configuration.

> .. note:: The Operator will not allow rescaling Percona Server for MongoDB with
>    the ``kubectl scale statefulset <StatefulSet name>`` command because this
>    would cause ``size`` configuration options to become out of sync.

Reviewer: ```suggestion
.. note:: The Operator will not allow to scale Percona Server for MongoDB with
   the ``kubectl scale statefulset <StatefulSet name>`` command as it puts
``size`` configuration options out of sync.
```

or down ensures new Pods are created and set to available Kubernetes
nodes.

The size of the cluster is controlled by the ``size`` key in the
:ref:`operator.custom-resource-options` configuration.

.. note:: Using ``kubectl scale StatefulSet_name`` command to rescale
   Percona Server for MongoDB is not recommended, as it makes ``size``
   configuration options out of sync, and the next config change may result

>    in reverting the previous number of nodes.

Reviewer: This is not correct.
1. Operator will not allow the user to scale statefulset and will revert to the original size in the next reconcile loop (5 seconds).
2. The command for scaling StatefulSet is: kubectl scale statefulset StatefulSet_name

or down ensures new Pods are created and set to available Kubernetes
nodes.

> The size of the cluster is controlled by ``size`` keys in the

Reviewer: Should it be ``size`` key, not key**s**?

.. note:: Using ``kubectl scale StatefulSet_name`` command to rescale
   Percona Server for MongoDB is not recommended, as it makes ``size``
   configuration options out of sync, and the next config change may result
   in reverting the previous number of nodes.

You can change size separately for different components of your cluster by
setting this option in the appropriate subsections:

* :ref:`replsets.size<replsets-size>` allows to set the size of the MongoDB
  Replica Set,
* :ref:`replsets.arbiter.size<replsets-arbiter-size> allows to set the number
  of :ref:`Replica Set Arbiter instances<arbiter>`,
* :ref:`sharding.configsvrReplSet.size<sharding-configsvrreplset-size>` allows
  to set the number of `Config Server instances <https://docs.mongodb.com/manual/core/sharded-cluster-config-servers/>`_,
* :ref:`sharding.mongos.size<sharding-mongos-size>` allows to set the number of `mongos <https://docs.mongodb.com/manual/core/sharded-cluster-query-router/>`_ instances.

For example, the following update in ``deploy/cr.yaml`` will set the size of the
MongoDB Replica Set to ``5`` nodes:

.. code:: yaml

   ....
   replsets:
     ....
     size: 5
     ....

> Don't forget to present changed configuration to the Operator as usual, running

Reviewer: Could you please review this line? 

"Don't forget to present changed configuration to the Operator as usual" sounds weird.

the ``kubectl apply -f deploy/cr.yaml`` command.

.. _operator.scale.scale-down:

Scaling the cluster down is done in a similar way as scaling it up. The only
thing to consider when you scale your cluster down is how the Operator treats
`Persistent Volume Claims <https://kubernetes.io/docs/concepts/storage/persistent-volumes/>`_.
By default, PVCs are kept after the database deletion. But you can enable
automatic removal of PVCs after the database scale down or deletion event by
adding the ``metadata.finalizers.delete-psmdb-pvc`` key in the

> ``deploy/cr.yaml`` configuration file.

Reviewer: @blez is scaling handled by finalizer in this release?

Reviewer 2: We have finalizer for cluster deletion, not scaling down
@spron-in
